lib/msal-common/src/client/OnBehalfOfClient.ts

```diff
@@ -267,6 +268,10 @@ export class OnBehalfOfClient extends BaseClient {
             parameterBuilder.addClientAssertionType(clientAssertion.assertionType);
         }
 
+        if (!StringUtils.isEmptyObj(request.claims) || this.config.authOptions.clientCapabilities && this.config.authOptions.clientCapabilities.length > 0) {
```

Comment:
This is ok, I won't block the PR on this and it'll fix the issue. We aren't making anymore releases this year. 

But @Robbie-Microsoft will need to follow up as there are several problems: 


- The request code is very WET. We duplicate this (and some other param setting) logic across confidential client. This issue is a good opportunity to make it DRY.
- if claims (but not client capabilities) are set, token cache needs to be bypassed (I'll log a new bug around this)

Reply:
@bgavrilMS I added tests to ensure that the claims are in the request body on the token acquisition and when configuring the `clientCapabilities`, but for other items happy to close this if another PR takes precedence.

Reply:
This is great, thanks @salman90